src/core/mac/mac.cpp

```diff
@@ -255,12 +250,12 @@ void Mac::PerformActiveScan(void)
 {
     if (UpdateScanChannel() == OT_ERROR_NONE)
     {
-        // If there are more channels to scan, start CSMA backoff to send the beacon request.
-        StartCsmaBackoff();
+        // If there are more channels to scan, send the beacon request.
+        BeginTransmit();
     }
     else
     {
-        otPlatRadioSetPanId(&GetInstance(), mPanId);
+        mSubMac.SetPanId(mPanId);
```

Comment:
There seems to be a mix of using `mPanId` and `GetPanId()` should we make this consistent?